### PR TITLE
Login

### PR DESCRIPTION
**Pull request by SerhiiYaroshenko1986**

### CODE REVIEW

File: dtApi/src/app/app.module.ts [hunk 1]
```diff
@@ -2,12 +2,23 @@ import { BrowserModule } from '@angular/platform-browser'
 import { NgModule } from '@angular/core'
 
 import { AppRoutingModule } from './app-routing.module'
+import { HttpClientModule } from '@angular/common/http'
+import { ReactiveFormsModule } from '@angular/forms'
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
```
Comment (vroh17) [inline dtApi/src/app/app.module.ts:5]: minor. try to group imports (angular, 3rd-party, your files) with empty line after each group

File: dtApi/src/app/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<div class="container">
+    <div class="form-wrap">
+        <!-- <div class="logo">
```
Comment (vroh17) [inline dtApi/src/app/login/login.component.html]: try to avoid comments

File: dtApi/src/app/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        this.badRequest = true
+        this.errorMessage = err.error.response
+        this.removeErrorMessage()
+    }
+    removeErrorMessage() {
+        setTimeout(() => {
+            this.badRequest = false
+        }, 1500)
+    }
+    ngOnInit(): void {}
```
Comment (vroh17) [inline dtApi/src/app/login/login.component.ts]: no need for empty hooks.
all component methods should go after Angular hooks. First public methods then private. The same for properties.

File: dtApi/src/app/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        this.loginForm.reset()
+        this.request.loginRequest(this.userName, this.password).subscribe(
+            (response) => {
+                // if (response.username === 'admin') {
+                //     this.router.navigate(['/admin/dashboard'])
+                // }
+                // if (response.username === 'student') {
+                //     this.router.navigate(['/student'])
+                // }
+                console.log(response)
```
Comment (vroh17) [inline dtApi/src/app/login/login.component.ts]: configure linter to not allow console.logs

File: dtApi/src/app/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+export class LoginComponent implements OnInit {
+    hide = true
+    badRequest = false
+    errorMessage: string
+    loginForm: FormGroup
+    private userName: string
+    private password: string
+
+    constructor(private request: AuthService, private router: Router) {
+        this._createForm()
```
Comment (vroh17) [inline dtApi/src/app/login/login.component.ts]: do not use _
```this._createForm()``` => ```this.createForm()```

File: dtApi/src/app/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    private _createForm() {
+        this.loginForm = new FormGroup({
```
Comment (vroh17) [inline dtApi/src/app/login/login.component.ts]: use FormBuilder for reactive forms
https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls

File: dtApi/src/app/login/services/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import { map, tap } from 'rxjs/operators'
+import { Router } from '@angular/router'
+
+@Injectable({
+    providedIn: 'root',
+})
+export class AuthService {
+    constructor(private http: HttpClient, private router: Router) {}
+    loginRequest(userName: string, password: string): Observable<any> {
+        const url = 'https://dtapi.if.ua/api/login'
```
Comment (vroh17) [inline dtApi/src/app/login/services/auth.service.ts]: move host url to app environments and import it from there

File: dtApi/src/app/login/services/auth.service.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        const body = {
+            username: userName,
+            password: password,
+        }
+        return this.http.post(url, body)
+    }
+    logOutrequest() {
```
Comment (vroh17) [inline dtApi/src/app/login/services/auth.service.ts]: typo. logOutrequest => logOutRequest

File: dtApi/src/app/login/services/auth.service.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+                    this.router.navigate(['/login'])
+                }
+            },
+            (err) => {
+                console.error(err)
+            }
+        )
+    }
+    isLogged() {
+        const url = 'https://dtapi.if.ua/api/login/isLogged'
```
Comment (vroh17) [inline dtApi/src/app/login/services/auth.service.ts]: create some config file with constants for API urls

**Comment by vroh17** [comment]
And I see your PR has conflicts. Resolve them first

File: dtApi/src/app/login/services/auth.service.ts (hunk 7 above)
Comment (yurkovskiy) [reply dtApi/src/app/login/services/auth.service.ts]: So, at environment folder you have two files 
- environment.ts
- environment.prod.ts
please put BASE URL https://dtapi.if.ua/api into these files

File: dtApi/src/app/login/services/auth.service.ts (hunk 9 above)
Comment (yurkovskiy) [reply dtApi/src/app/login/services/auth.service.ts]: This constant should be like this
const URI = `$BASEURL/......`

**Review by vroh17 — DISMISSED** [review]

**Review by KvakSebastian — APPROVED** [review]

**Review by MaaxLav — APPROVED** [review]

**Review by olehyaremyn7 — APPROVED** [review]

**Pull request merged**